.github/workflows/scripts/e2e-verify.common.sh

```diff
@@ -79,21 +79,29 @@ verify_provenance_authenticity() {
     local annotated_tags
     annotated_tags=$(echo "$THIS_FILE" | cut -d '.' -f5 | grep annotated || true)
 
-    # TODO: Currently we only support $BINARY artifacts, not containers.
     verifierCmd="$verifier"
+    # After version 1.3.0, we split into subcommands for artifacts and images
     if [[ "$tag" == HEAD ]] || version_gt "$tag" "v1.3.0"; then
-        verifierCmd="$verifier verify-artifact"
+        if [[ -n $BINARY ]]; then
+            verifierCmd="$verifier verify-artifact"
+        elif [[ -n $CONTAINER ]]; then
+            verifierCmd="$verifier verify-image"
+        fi
     fi
     # This transforms the argument name depending on the verifier tag.
     read -ra argr <<<"$(e2e_verifier_arg_transformer "$tag")"
```

Comment:
I realize you updated this deliberately in a previous patch but I wonder if that's right? IIUC `e2e_verifier_arg_transformer` returns a function name and not a space delimited string that we need to expand into an array.

The warnings below are a result of argr being an array so I think the type is incorrect here.

*God, bash is so confusing...

Reply:
you are right here!